### PR TITLE
log-backup: make the download more finer-grained

### LINKED ISSUE #14206: log-backup: get rid of the dashmap locks
## Development Task
Dashmap uses segment lock for locking a shard of entries. Which limits the max concurrency of downloading, we need get rid of it.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: Close #14206

What's Changed:
This PR make the downloaded cache file hold lock for itself. So we don't need to rely on the segment lock of the dashmap.

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test (TBD)
- Manual test (add detailed scripts or steps below)

### Release note 

```release-note
Speeded up PITR.
```

### CODE REVIEW

File: components/sst_importer/src/sst_importer.rs [hunk 1]
```diff
@@ -490,21 +662,41 @@ impl SstImporter {
...
+        let promise = {
+            let lock = self.file_locks.entry(dst_name);
+            IMPORTER_APPLY_DURATION
+                .with_label_values(&["download-get-lock"])
+                .observe(start.saturating_elapsed().as_secs_f64());
+
+            match lock {
+                Entry::Occupied(mut ent) => match ent.get_mut() {
+                    (CacheKvFile::Mem(buff), last_used) => {
+                        *last_used = Instant::now();
```
Comment (3pointer) [inline components/sst_importer/src/sst_importer.rs:674]: BTW what's the usage of `last_used`?

**Review by 3pointer — APPROVED** [review]

File: components/sst_importer/src/sst_importer.rs (hunk 1 above)
Comment (YuJuncen) [reply components/sst_importer/src/sst_importer.rs:674]: We would check it in interval and remove outdated cache according to it.

File: components/sst_importer/src/sst_importer.rs [hunk 2]
```diff
@@ -452,7 +624,6 @@ impl SstImporter {
             if shrink_buff_size > 0 || retain_buff_size > 0 {
                 info!("shrink cache by tick"; "shrink size" => shrink_buff_size, "retain size" => retain_buff_size);
             }
-            self.dec_mem(shrink_buff_size as _);
```
Comment (joccau) [inline components/sst_importer/src/sst_importer.rs:455]: If remove here, how to statistic the  released memory if remove the time.

**Review by joccau — COMMENTED** [review]
rest LGTM

File: components/sst_importer/src/sst_importer.rs (hunk 2 above)
Comment (joccau) [reply components/sst_importer/src/sst_importer.rs:455]: OK, I get it~

**Review by joccau — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

**Pull request merged**